pkg/filters/builder/requestadaptor.go

```diff
@@ -125,21 +126,23 @@ var signerConfigs = map[string]signerConfig{
 }
 
 func init() {
-	filters.Register(kind)
+	filters.Register(requestAdaptorKind)
 }
 
 type (
 	// RequestAdaptor is filter RequestAdaptor.
 	RequestAdaptor struct {
-		spec *Spec
+		spec *RequestAdaptorBuilderSpec
+		Builder
 
 		pa     *pathadaptor.PathAdaptor
 		signer *signer.Signer
 	}
 
-	// Spec is HTTPAdaptor Spec.
-	Spec struct {
+	// RequestAdaptorBuilderSpec is HTTPAdaptor RequestAdaptorBuilderSpec.
+	RequestAdaptorBuilderSpec struct {
```

Comment:
why rename this structure?

Reply:
Because it named Spec which confilict with ResponseAdaptor which also named Spec.

Reply:
But we can name it `RequestAdaptorSpec`, as all other filters.

Reply:
According to the [samanhappy](https://github.com/samanhappy),
> we can move requestadaptor and responseadaptor files to the builder directory directly for now to avoid renaming too many variable and method names, and we can do it in the future refactor work if necessary

I moved requestadaptor and responseadaptor files to the builder directory directly, so it comes conflict with the same name `Spec`.

Reply:
Actually some private fields if it has are not needed by the template indeed.  Define the template as a standalone structure and copy the definition of the common fields  may add redundancy to the code. Actually the template definition is the base structure to the `spec` for `RequestAdaptor` and `ResponseAdaptor`. If we would like to define a standalone structure for the template, I prefer `RequestAdaptorBuilderSpec` and `ResponseAdaptorBuilderSpec` inheriting the template structure to avoid redefinition such as
```golang
		Host       string                `json:"host" jsonschema:"omitempty"`
		Method     string                `json:"method" jsonschema:"omitempty,format=httpmethod"`
		Path       *pathadaptor.Spec     `json:"path,omitempty" jsonschema:"omitempty"`
		PathStr    string                `json:"pathStr,omitempty" jsonschema:"omitempty"`
		Header     *httpheader.AdaptSpec `json:"header,omitempty" jsonschema:"omitempty"`
		Body       string                `json:"body" jsonschema:"omitempty"`
		Compress   string                `json:"compress" jsonschema:"omitempty"`
		Decompress string                `json:"decompress" jsonschema:"omitempty"`
		Sign       *SignerSpec           `json:"sign,omitempty" jsonschema:"omitempty"`
```





> hi @hhhcj , after think it twice, I think your comments make sense.
> however, please don't use the spec of the filter as the template, as many fields of the spec are not needed by the template. the better way is to define the template as a standalone structure and embed it into the spec.

Reply:
For me the request and response are different, we better not to couple them together, but the template spec and static spec are similar and we can use a inheritance for them like 

> define the template as a standalone structure and embed it into the spec.